xain_fl/coordinator/metrics_store.py

```diff
@@ -132,44 +166,104 @@ def write_participant_metrics(self, metrics_as_json: str) -> None:
         try:
             metrics = json.loads(metrics_as_json)
             validate(instance=metrics, schema=self.schema)
-            self.influx_client.write_points(metrics)
-        except Exception as err:  # pylint: disable=broad-except
+        except (ValidationError, JSONDecodeError) as err:
             logger.error("Exception", error=repr(err))
             raise MetricsStoreError("Can not write participant metrics.") from err
+        else:
+            self._write_metrics(metrics)
+
+    def write_participant_metrics(
```

Comment:
Can we unify write_coordinator_metrics and write_participant_metrics to avoid code duplication, the code is basically the same except the value of one parameter ("coordinator" or "participant"), which could be a parameter of the function. What do you think?

Reply:
Thanks for the review👍
In my option, both methods make the code simpler and less error prone. Since the parameter is a string, it is prone to typos. Another downside is that I have to add `coordinator` or `participant` to each `metrics_store.write_metrics` call in the `coordinator.py` later which can be seen as code duplication too. 
But I see your point, I will remove both methods and just keep the `write_metrics` methods.

Reply:
I applied your requested changes.